From 4f302276acc9a23bfc216fb18a2f30d150db8f30 Mon Sep 17 00:00:00 2001
From: Steven Weaver <sweaver@temple.edu>
Date: Sat, 28 Oct 2023 22:03:10 -0600
Subject: [PATCH] use emmake

This patch transitions the build process in `setup.py` to use `emmake`,
a utility provided by Emscripten that wraps make commands to ensure that
the correct flags are passed to the Emscripten compiler.

Changes introduced:
- The `run_make` and `run_make_print_config` functions are modified to
  invoke `emmake` before the system's `make` command.
- This adjustment ensures that the `make` utility operates with the
  appropriate environment variables and configurations required for
  Emscripten to compile C and C++ to WebAssembly.


---
 setup.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/setup.py b/setup.py
index 22f3cb3..39da16d 100644
--- a/setup.py
+++ b/setup.py
@@ -75,11 +75,11 @@ def run_configure(option):
 
 def run_make(targets):
     sys.stdout.flush()
-    subprocess.check_call([os.environ.get("MAKE", "make")] + targets)
+    subprocess.check_call(["emmake", os.environ.get("MAKE", "make")] + targets)
 
 
 def run_make_print_config():
-    stdout = subprocess.check_output(["make", "-s", "print-config"])
+    stdout = subprocess.check_output(["emmake", "make", "-s", "print-config"])
     if IS_PYTHON3:
         stdout = stdout.decode("ascii")
 
-- 
2.39.3 (Apple Git-145)

